test/parallel/test-process-constants-noatime.js

```diff
@@ -2,10 +2,11 @@
 
 require('../common');
 const assert = require('assert');
-
-const isLinux = process.platform === 'linux';
-
-const O_NOATIME = process.binding('constants').O_NOATIME;
-const expected = isLinux ? 0x40000 : undefined;
-
-assert.strictEqual(O_NOATIME, expected);
+const constants = process.binding('constants');
+
+if (process.platform === 'linux') {
+  assert(constants.hasOwnProperty('O_NOATIME'));
+  assert.strictEqual(constants.O_NOATIME, 0x40000);
+} else {
+  assert(false === constants.hasOwnProperty('O_NOATIME'));
```

Comment:
If you use `!('O_NOATIME' in constants)`, it will check the prototype as well (for the very unlikely case that we decide to move things to the prototype in the future.)